File: Python/백준/Bronze/31067. 다오의 경주 대회/다오의 경주 대회.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
tracks = list(map(int, input().split()))
count = 0

for i in range(1, n):
	if tracks[i-1] >= tracks[i]:
		if tracks[i-1]-tracks[i] < k:
			count += 1
			tracks[i] += k

		else:
			print(-1)
			sys.exit()

print(count)

File: Java/백준/Bronze/31067. 다오의 경주 대회/다오의 경주 대회.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        long n = Long.parseLong(st.nextToken());
        long k = Long.parseLong(st.nextToken());

        long[] tracks = new long[(int) n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            tracks[i] = Long.parseLong(st.nextToken());
        }

        int count = 0;

        for (int i = 1; i < n; i++) {
            if (tracks[i-1] >= tracks[i]) {
                if (tracks[i-1]-tracks[i] < k) {
                    count++;
                    tracks[i] += k;
                }
                else {
                    System.out.println(-1);
                    System.exit(0);
                }
            }
        }
        System.out.println(count);
    }
}